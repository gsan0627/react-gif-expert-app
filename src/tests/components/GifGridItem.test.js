import React from "react";
const { shallow } = require("enzyme");
const { GifGridItem } = require("../../components/GifGridItem");

describe("testing on <GifGridItem />", () => {
  const title = "Dragon Ball Z";
  const url = "https://localhost/dbz.png";
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test("should display the component correctly ", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("must have a paragraph with the title", () => {
    const p = wrapper.find("p");
    expect(p.text().trim()).toBe(title);
  });

  test("must have the image equal to the url and alt of the props", () => {
    const img = wrapper.find("img");
    expect(img.prop("src")).toBe(url);
    expect(img.prop("alt")).toBe(title);
  });

  test("must have animate__bounce", () => {
    const div = wrapper.find("div");
    const className = div.prop("className");
    expect(className.includes("animate__bounce")).toBe(true);
  });
});
