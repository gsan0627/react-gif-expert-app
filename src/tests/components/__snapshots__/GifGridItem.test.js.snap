// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing on <GifGridItem /> should display the component correctly  1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="Dragon Ball Z"
    src="https://localhost/dbz.png"
  />
  <p>
    Dragon Ball Z
  </p>
</div>
`;
